site/content/blog/2018-03-12-GA-GTM.adoc

```diff
@@ -0,0 +1,90 @@
+Cette information se trouve normalement dans la variable `Page Path` qui fait partie des variables activées par défaut à la création du conteneur.
+Mais si comme nous vous faites beaucoup d'*Angular*, vous avez à faire avec des urls du type `http://localhost:9000/#/home`.
+Vous savez que ce n'est pas aussi simple et vous avez déjà dû vous retrouver à mettre ce genre de code pour traquer exactement la page visualisée et pas tout le temps `/`.
+
+[source, javascript]
+----
+  this.router.events.subscribe((event) => {
+            if (event instanceof NavigationEnd) {
+                ga('set', 'page', event.urlAfterRedirects);
+                ga('send', 'pageview');
+            }
+        });
+----
+
+Pour répondre à ce problème avec *Google Tag Manager*, il faut activer une nouvelle variable `New History Fragment` se basant sur l'API History du navigateur pour stocker la valeur de la nouvelle page.
+Pour faire cela, vous avez juste à cliquer sur `Variables` dans le menu, puis `CONFIGURER` et enfin cocher `New History Fragment`.
+
+Afin d'envoyer un nouvel évènement à chaque changement de page, nous devons déclarer un nouveau déclencheur se basant sur l'API History. Pour cela, cliquez sur le menu `Déclencheurs`, puis `NOUVEAU`.
+À ce moment vous devez cocher l'item `History Change` dans la liste s'affichant lorsque vous cliquez dans le cadre blanc.
+
+Enfin, vous n'avez plus qu'à créer votre balise `Analytics`.
+Pour cela rendez-vous dans le menu `Balises`, lorsque vous cliquez sur `NOUVELLE`, vous pouvez sélectionner le type de balise `Universal Analytics` dans le premier cadre et votre déclencheur préalablement créé dans le deuxième.
+Il reste deux informations à configurer, le tracking id que vous pouvez ajouter en configurant une nouvelle variable dans `Google Analytics Settings` et la modification pour prendre en compte votre variable se basant sur l'API History.
+Il faut cocher `Enable overriding settings in this tag`, déplier `More settings` et `Fields to Set` et ajouter un champ `page` correspondant à la valeur `{{New History Fragment}}` (nom de votre variable créée).
+
+À partir de là notre conteneur est bien configuré il n'y a plus qu'à tester et livrer.
+ Pour ce faire, vous devez activer le mode debug en cliquant sur `PRÉVISUALISER` si ce n'est pas déjà fait et lancer votre application.
+ Dans le panel de debug, vous verrez normalemement la liste des différents `déclencheurs` sur la gauche, et lorsque vous cliquez dessus, vous pouvez voir quelles balises ont été activées et dans l'onglet `Variables`, la valeur des différentes variables (dont votre `New History Fragment`).
+
+:lightboxGroup: 0
+:lightboxImage: 2018-03-22_GA-GTM/debug_panel_1
+:lightboxImageTitle: Panel de debug
+
+include::site/static/lightbox.adoc[]
+
+Il ne reste plus qu'à publier vos modifications pour qu'elles soient actives sur tous vos environnements en cliquant sur `ENVOYER` et `PUBLIER`.
```

Comment:
There is a lot of mix between french and english wording in this block, we should maybe put everything in english ?

Reply:
This is what we have on google tag manager dashboard (if the browser is in french). Variables are always in english.

Reply:
Looks ok then :)